CONTRIBUTING.md

```diff
@@ -71,6 +71,54 @@ $ git rebase master
 
 > Now your `feature/my-new-feature` branch is up-to-date with all the code in `upstream/master`.
 
+Running E2E Tests locally before commits
+----------------------------------------
+E2E Tests can be run against the Custom Login, Okta-Hosted Login and Resource servers
+
+Follow the steps below to run the tests locally:
+
+```bash
+# At project root
+composer install
+```
+To test the samples you will need the following configured in your developer org:
+
+* [A Web application](/okta-hosted-login#prerequisites)
+* [A SPA application](https://github.com/okta/samples-js-angular/tree/master/okta-hosted-login#prerequisites)
+* A test user account with a known username and password. Note that the USERNAME should be of the form "username@email.com"
+
+Once you have those resources setup, export their details as the following environment variables:
+
+```bash
+export ISSUER=https://{yourOktaDomain}.com/oauth2/default
+export CLIENT_ID={yourWebAppClientId}
+export CLIENT_SECRET={yourWebAppClientSecret}
+export SPA_CLIENT_ID={yourSpaAppClientId}
```

Comment:
Do we actually need a spa client id?  I was able to do all of this with the same client_id / application as my web side app in testing.

Reply:
For resource server, we need an SPA access token. If I use the same client_id in `.env` of resource-server, the tests fail.